add-on/src/types.d.ts

```diff
@@ -0,0 +1,52 @@
+
```

Comment:
Typescript types supported for linting. No work done for building validation (except from linter) nor exporting types (shouldn't be needed).

Errors show in vscode but don’t fail build yet